runtime/lua/vim/iter.lua

```diff
@@ -112,6 +112,35 @@ local function sanitize(t)
   return t
 end
 
+---Flattens a single list-like table. Errors if it attempts to flatten a
+---dict-like table
+---@param v table table which should be flattened
+---@param max_depth number depth to which the table should be flattened
+---@param initial_depth number|nil (internal) starting depth (defaulted to 0)
+---@param result table|nil (internal) flattened in each iteration (defaulted to {})
+---@return table flattened table if it can be flattened, otherwise nil
+local function flatten_single_list(v, max_depth, initial_depth, result)
+  local d = initial_depth or 0
+  local res = result or {}
+
+  if d < max_depth and type(v) == 'table' then
+    local i = 0
+    for _ in pairs(v) do
+      i = i + 1
+      -- short-circuit: this is not a list like table
+      if v[i] == nil then
+        error('list like table encountered')
```

Comment:
The error message doesn't seem to match the comment? Should this say "non-list like table encountered"?

There should be a test case covering this scenario (that matches on the error text).

Reply:
this is the error text in the helper method - it isn't surfaced to the actual method. there is a test case that covers this scenario: https://github.com/neovim/neovim/pull/26786/files#diff-23eaf1539c9c72672e24dc1ba1cbe1e57dc45822d8132408e12b54c76ff4a20fR420

Reply:
Is the error text correct? It doesn't look like it -- as I said above, the comment seems to contradict the error text.

Reply:
ohh I see what you're saying. yes, this error text is incorrect even though it isn't what is being returned to the user. I will fix that